src/routes/blog/_query.ts

```diff
@@ -103,6 +113,14 @@ export async function fetchNewsletters(): Promise<Newsletter[]> {
     body: JSON.stringify({ query: newslettersQuery }),
   });
 
-  const newsletters = formatNewsletters(await response.json());
-  return newsletters;
+  let newsletters = formatNewsletters(await response.json());
+  const tags = discernTags(newsletters);
+
+  if (options && options.labels.length > 0) {
+    newsletters = newsletters.filter((post) =>
```

Comment:
Do not forget to swap out the variable names in this line.
```suggestion
    posts = posts.filter((post) =>
```